crates/sparrow-core/src/lib.rs

```diff
@@ -11,12 +11,14 @@ use std::pin::Pin;
 
 mod context_code;
 mod key_triple;
+mod read_config;
 mod table_schema;
 
 use arrow::record_batch::RecordBatch;
 pub use context_code::*;
 use futures::Stream;
 pub use key_triple::{KeyTriple, KeyTriples};
+pub use read_config::ReadConfig;
```

Comment:
Should `ReadConfig` live in interfaces? What needs it without needing the interfaces (eg., why should it be in `core`)?

Ideally, we'd be moving things out of `sparrow-core` into more appropriate places where possible.

Reply:
Hm..likely nothing at this point, though I could see a world where we create the `ReadConfig` in `session` (parameters passed from the client), then just passed through until they're used in `sparrow-execution` (or wherever `read/execute` is called)

Reply:
But session will depend on `sparrow-interfaces` so it can do `add_source(source: Arc<dyn Source>)` right? Which means moving `ReadConfig` to `sparrow-interfaces` would meanthat session doesn't need to depend on `sparrow-core` right?

Reply:
`ReadConfig` would theoretically be unpacked into its component parts, so it makes sense that it could live within the `read` method. `source.rs` trait.